apps/twitter-auth/src/app/app.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Req, Res, UnauthorizedException, UseGuards } from '@nestjs/common';
+import { Controller, Get } from '@nestjs/common';
+import { Request, Response } from 'express';
+import { TwitterAuthGuard } from './passport/twitter/twitter-auth.guard';
+import { PassportSessionGuard } from './passport/passport-session.guard';
+
+@Controller('auth')
+export class AppController {
+  @Get('twitter')
+  @UseGuards(TwitterAuthGuard)
+  async twitter() {
+    throw new UnauthorizedException(); // Guard redirects
+  }
+
+  @Get('twitter/callback')
+  @UseGuards(TwitterAuthGuard)
+  async twitterCallback(@Req() req: Request, @Res() res: Response) {
+    // Redirect the callback from Twitter to an Passport-session protected route
+    res.redirect('test');
+  }
```

Comment:
As discussed on Discord, we may want to make this an HTML string or similar and make the user navigate to the `test` endpoint themselves to ensure the session token gets initialized and saved properly.

```suggestion
  @Get('twitter/callback')
  @UseGuards(TwitterAuthGuard)
  async twitterCallback(@Req() req: Request) {
    console.log('In twitter callback')
    // Redirect the callback from Twitter to an Passport-session protected route
    return '<a href="test">Click Me!</a>';
  }
```